src/main/java/ch/njol/skript/lang/function/DynamicFunctionReference.java

```diff
@@ -56,11 +57,16 @@ public DynamicFunctionReference(@NotNull String name, @Nullable Script source) {
 		this.name = name;
 		Function<? extends Result> function;
 		if (source != null) {
+			// will return the first function found that matches name.
+			// TODO: add a way to specify param types
 			//noinspection unchecked
-			function = (Function<? extends Result>) Functions.getFunction(name, source.getConfig().getFileName());
+			function = (Function<? extends Result>) FunctionRegistry.getRegistry().getFunction(source.getConfig().getFileName(), name);
 		} else {
 			//noinspection unchecked
-			function = (Function<? extends Result>) Functions.getFunction(name, null);
+			function = (Function<? extends Result>) FunctionRegistry.getRegistry().getFunction(null, name);
+		}
+		if (function == null) {
+			Skript.error(FunctionReference.AMBIGUOUS_ERROR.formatted(name));
 		}
```

Comment:
I don't think we want to be leaving an error here. Looking at the usages, it's always created at runtime, and there is no point in erroring then (unless it's a runtime error, but I do not think that's applicable). I think you're going to need more extensive changes here to resolve the function at runtime based on the passed parameters (consider dynamic function execution via syntax)